### PR TITLE
Server side checksum verification

### PR DESCRIPTION
**Pull request by MorrisJobke**
* downstream of https://github.com/owncloud/core/pull/27097

### CODE REVIEW

**Comment by MorrisJobke** [comment]
We maybe put also https://github.com/owncloud/core/pull/27345 in this PR

File: apps/files/lib/Capabilities.php [hunk 1]
```diff
@@ -53,6 +53,10 @@ public function __construct(IConfig $config) {
 	 */
 	public function getCapabilities() {
 		return [
+			'checksums' => [
+				'supportedTypes' => ['SHA1'],
```
Comment (rullzer) [inline apps/files/lib/Capabilities.php]: maybe we should also offer something else :wink:

**Comment by rullzer** [comment]
Just to explain. This is not checksum verification as far as I can tell from a quick look.
It is still using the stored checksum on the server.

**Review by icewind1991 — CHANGES_REQUESTED** [review]
Let me fix this up a bit before merging

**Comment by MorrisJobke** [comment]
> Let me fix this up a bit before merging

Tests are still failing

**Comment by MorrisJobke** [comment]
@icewind1991 Could you have a look at the tests?

**Comment by MorrisJobke** [comment]
> @icewind1991 Could you have a look at the tests?

Ping

**Comment by MorrisJobke** [comment]
Continued in #4520

**Pull request closed without merging**